tests/test_casefold_migration.py

```diff
@@ -0,0 +1,311 @@
+
+        # create some global associations
+        associations = []
+        originServer = self.sydent.server_name
+
+        for i in range(10):
+            address = "bob%d@example.com" % i
+            mxid = "@bob%d:example.com" % i
+            ts = 10000 * i
+            associations.append(
+                {
+                    "medium": "email",
+                    "address": address,
+                    "lookup_hash": calculate_lookup_hash(self.sydent, address),
+                    "mxid": mxid,
+                    "ts": ts,
+                    "not_before": 0,
+                    "not_after": 99999999999,
+                    "originServer": originServer,
+                    "originId": i,
+                    "sgAssoc": json.dumps(
+                        self.create_signedassoc(
+                            "email", address, mxid, ts, 0, 99999999999
+                        )
+                    ),
+                }
+            )
+        # create some casefold-conflicting associations
+        for i in range(5):
+            address = "BOB%d@example.com" % i
+            mxid = "@BOB%d:example.com" % i
+            ts = 10000 * i
+            associations.append(
+                {
+                    "medium": "email",
+                    "address": address,
+                    "lookup_hash": calculate_lookup_hash(self.sydent, address),
+                    "mxid": mxid,
+                    "ts": ts + 1,
+                    "not_before": 0,
+                    "not_after": 99999999999,
+                    "originServer": originServer,
+                    "originId": i + 10,
+                    "sgAssoc": json.dumps(
+                        self.create_signedassoc(
+                            "email", address, mxid, ts, 0, 99999999999
+                        )
+                    ),
+                }
+            )
+
+        # add all associations to db
+        cur = self.sydent.db.cursor()
+
+        cur.executemany(
+            "INSERT INTO global_threepid_associations "
+            "(medium, address, lookup_hash, mxid, ts, notBefore, notAfter, originServer, originId, sgAssoc) "
+            "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+            [
+                (
+                    assoc["medium"],
+                    assoc["address"],
+                    assoc["lookup_hash"],
+                    assoc["mxid"],
+                    assoc["ts"],
+                    assoc["not_before"],
+                    assoc["not_after"],
+                    assoc["originServer"],
+                    assoc["originId"],
+                    assoc["sgAssoc"],
+                )
+                for assoc in associations
+            ],
+        )
+
+        self.sydent.db.commit()
+
+    def test_migration_email(self):
+        with patch("sydent.util.emailutils.smtplib") as smtplib:
+            templateFile = self.sydent.get_branded_template(
+                "matrix-org",
```

Comment:
```suggestion
               None,
```

Just so it's consistent with the script (it probably doesn't matter that much since the default config, which is also used for tests, has `matrix-org` as its default branch, but it'll save us some trouble if we want to change that default in the future).